src/pages/studyView/StudyViewUtils.tsx

```diff
@@ -977,6 +986,55 @@ export function calculateLayout(visibleAttributes: ChartMeta[], cols: number): L
     return layout;
 }
 
+export function calculateNewLayoutForFocusedChart(previousLayout: Layout, currentFocusedChartByUser: ChartMeta, cols: number): Layout {
+    const initialX = previousLayout.x;
+    const initialY = previousLayout.y;
+    const dimensionX = currentFocusedChartByUser.dimension.w;
+    let x = initialX;
+    let y = initialY;
+
+    if (isFocusedChartShrinked({w: previousLayout.w, h: previousLayout.h}, currentFocusedChartByUser.dimension)) {
+        x = initialX + previousLayout.w - dimensionX;
+    }
+    else if (initialX + dimensionX >= cols && initialX - dimensionX >= 0) {
+        x = cols - dimensionX;
+    }
+
+    return {
+        i: previousLayout.i,
+        x,
+        y,
+        w: currentFocusedChartByUser.dimension.w,
+        h: currentFocusedChartByUser.dimension.h,
+        isResizable: false
+    };
+}
+
+export function populateFixedPositionLayout(layout: Layout, matrix: string[][], cols: number) {
+    const xMax = layout.x + layout.w;
+    const yMax = layout.y + layout.h;
+    while (yMax >= matrix.length) {
+        matrix.push(new Array(cols).fill(''));
+    }
+    for (let i = layout.y; i < yMax; i++) {
+        for (let j = layout.x; j < xMax; j++) {
+            matrix[i][j] = layout.i!;
+        }
+    }
+}
+
+export function isFocusedChartShrinked(oldDimension: ChartDimension, newDimension: ChartDimension): boolean {
+    return (oldDimension.w + oldDimension.h) > (newDimension.w + newDimension.h) ? true : false;
+}
+
+export function getPositionXByUniqueKey(layout: Layout[], uniqueKey: string): number {
+    return (_.find(layout, {'i': uniqueKey}) as Layout).x;
```

Comment:
These two functions have to do undefined check. The layout does not necessary have uniqueKey. This is why you need to do a force type here I think.